sktime/transformations/panel/dictionary_based/_sfa.py

```diff
@@ -459,51 +536,107 @@ def _discrete_fourier_transform(self, series):
-    def _mft(self, series, transformed=None, stds=None):
-        """
+        Input
+        -------
+        X : The training input samples.  array-like or sparse matrix of
+        shape = [n_samps, num_atts]
 
-        :param series:
-        :return:
+        Returns
+        -------
+        1D array of fourier term, real_0,imag_0, real_1, imag_1 etc, length
+        num_atts or
+        num_atts-2 if if self.norm is True
         """
-        start_offset, length, end = self._mft_start_length_end(series)
+        start = 2 if norm else 0
+        output_length = start + dft_length
+
+        if cut_start_if_norm:
+            c = int(start / 2)
+            dft = np.zeros(output_length - start)
+            for i in range(c, int(output_length / 2)):
+                for n in range(len(series)):
+                    dft[(i - c) * 2] += series[n] * math.cos(
+                        2 * math.pi * n * i / len(series)
+                    )
+                    dft[(i - c) * 2 + 1] += -series[n] * math.sin(
+                        2 * math.pi * n * i / len(series)
+                    )
+        else:
+            dft = np.zeros(output_length)
+            for i in range(int(output_length / 2)):
+                for n in range(len(series)):
+                    dft[i * 2] += series[n] * math.cos(
+                        2 * math.pi * n * i / len(series)
+                    )
+                    dft[i * 2 + 1] += -series[n] * math.sin(
+                        2 * math.pi * n * i / len(series)
+                    )
 
-        phis = np.array(
-            [
-                [
-                    math.cos(2 * math.pi * (-i) / self.window_size),
-                    -math.sin(2 * math.pi * (-i) / self.window_size),
-                ]
-                for i in range(0, int(length / 2))
-            ]
-        ).flatten()
+        if apply_normalising_factor:
+            if lower_bounding:
+                dft[1::2] = dft[1::2] * -1  # lower bounding
 
-        if stds is None or np.shape(stds) != (end):
-            stds = np.zeros(end)
+            std = np.std(series)
+            if std == 0:
+                std = 1
+            dft *= inverse_sqrt_win_size / std
 
-        stds = SFA._calc_incremental_mean_std(series, end, self.window_size, stds)
+        return dft
 
-        if transformed is None or np.shape(transformed) != (end, length):
-            transformed = np.zeros((end, length))
+    def _mft(self, series):
+        """
+
+        :param series:
+        :return:
+        """
+        start_offset = 2 if self.norm else 0
+        length = self.dft_length + start_offset + self.dft_length % 2
+        end = max(1, len(series) - self.window_size + 1)
+
+        phis = SFA._get_phis(self.window_size, length)
+        stds = SFA._calc_incremental_mean_std(series, end, self.window_size)
+        transformed = np.zeros((end, length))
+
+        # first run with dft
+        if self.fourier_transform == "dft":
```

Comment:
Not sure, if this is correct?

The first run of the mft is always the fft. and only for the subsequent runs it should use the mft-trick.

Howerver, given this switch, it will either always use the fft or never?

Reply:
Im a little confused at which part you are talking about. The the first run is always the DFT/FFT and subsequent ones use the iterate MFT function regardless of the setting.

Reply:
Sorry, you are absolutely right :) My comment was wrong. There is still the `SFA._iterate_mft` further down, that does exactly this. Sorry.

So, one uses the explicit implementation while the other one used the rfft.

But inverse_sqrt_win_size is also applied some lines further down in 635?

Reply:
DFT parameter `apply_normalising_factor=False` prevents the application twice.
Output required is slightly different for binning and mft so there is also `cut_start_if_norm=False` for DFT.

Reply:
Ah, ok, that's fine. Missed that part.

But maybe you could shorten the     `def _discrete_fourier_transform(` function using:

        if cut_start_if_norm:
            c = int(start / 2)
        else:
            c = 0
            start = 0

and only write the transformation code once?